src/bundler.ts

```diff
@@ -293,24 +321,23 @@ class Bundler {
         importParent = htmlImport.parentNode!;
       }
 
-      dom5.queryAll(importDoc, matchers.htmlImport).forEach((nestedImport) => {
-        this.inlineHtmlImport(
-            docUrl, nestedImport, importMap, bundle, manifest);
-      });
+      const nestedImports = dom5.queryAll(importDoc, matchers.htmlImport);
+      for (const nestedImport of nestedImports) {
+        await this.inlineHtmlImport(
+            docUrl, nestedImport, reachedImports, bundle, manifest);
+      }
 
       astUtils.insertAllBefore(importParent, htmlImport, importDoc.childNodes!);
     }
 
-    // If we've just inlined it or otherwise have seen it before, we can remove
-    // the <link> tag.
-    if (importMap.get(resolvedUrl) === null) {
+    if (reachedImports.has(resolvedUrl)) {
       dom5.remove(htmlImport);
     }
```

Comment:
Should the following `if (!reachedImports.has(resolvedUrl)` maybe be converted to an else clause of this if statement?